samcli/commands/_utils/click_mutex.py

```diff
@@ -0,0 +1,80 @@
+"""
+Module to check mutually exclusive cli parameters
+"""
+from typing import Any, List, Dict, Tuple
+
+import click
+
+
+class ClickMutex(click.Option):
+    """
+    Preprocessing checks for mutually exclusive or required parameters as supported by click api.
+
+    required_param_lists: List[List[str]]
+        List of lists with each supported combination of params
+        Ex:
+        With option = "a" and required_param_lists = [["b", "c"], ["c", "d"]]
+        It is valid to specify --a --b --c or --a --c --d
+        but not --a --b --d
+
+    required_params_hint: str
+        String to be appended after default missing required params prompt
+
+    incompatible_params: List[str]
+        List of incompatible parameters
+
+    incompatible_params_hint: str
+        String to be appended after default incompatible params prompt
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.required_param_lists: List = kwargs.pop("required_param_lists", [])
+        self.required_params_hint: str = kwargs.pop("required_params_hint", "")
+        self.incompatible_params: List = kwargs.pop("incompatible_params", [])
+        self.incompatible_params_hint: str = kwargs.pop("incompatible_params_hint", "")
+
+        super().__init__(*args, **kwargs)
+
+    def handle_parse_result(self, ctx: click.Context, opts: Dict[str, Any], args: List[str]) -> Tuple[Any, List[str]]:
+        if self.name not in opts:
```

Comment:
Regarding to our discussion below, I feel like having a small doc about this method and its parameters could be useful for the future maintenance.

Reply:
Added method docs.